contents/docs/contribute/coding-conventions/index.md

```diff
@@ -51,8 +51,47 @@ Don’t log sensitive information. Make sure you never log:
 #### Unit tests
 A good unit test should:
 * focus on a single use-case at a time
-* have a minimal set of assertion per test
+* have a minimal set of assertions per test
 * demonstrate every use case. The rule of thumb is: if it can happen, it should be covered
 
 #### Integration tests
 Integration tests should ensure that the feature works end-to-end. They must cover all the important use cases of the feature.
+
+### Frontend 
+
+#### Two layers: Kea -> React
+
+Our frontend webapp is written with [Kea](https://keajs.org/) and [React](https://reactjs.org/) as two separate layers. Kea is used to organise the app's data for rendering (we call this the *data* or *state* layer), and React is used to render the computed state (this is the *view* or *template* layer).
+
+We try to be very explicit about this separation, and avoid local React state wherever possible, with exceptions for the `lib/` folder. Having all our data in one layer makes for code that's easier to [test](https://kea.js.org/docs/guide/testing), and observe. Basically, getting your [data layer](https://kea.js.org/blog/data-first-frontend-revolution) right is hard enough. We aim to not make it harder by constraining your data to a DOM-style hierarchy.
+
+Hence the explicitly in keeping the layers separate.
+
+#### General tips
+
+- Think data first: get [your mental model of the data flowing through the app](https://acco.io/i-escaped-node) right, and then everything else will be easy.
+- Be practical, yet remember that you are balancing speed of delivery with ease of maintainability. If you have to choose: code should be easier to understand than it was to write.
+
+#### Do-s & Don't-s
+
+- General
+  - Write all new code with TypeScript and proper typing.
+  - Write your frontend data handling code first, and write it in a Kea `logic`.
+  - Don't use `useState` or `useEffect` to store local state. It's false convenience. Take the extra 3 minutes and change it to a `logic` early on in the development.
+  - Logics still have a tiny initialization cost. Hence this rule doesn't apply to library components in the `lib/` folder, which might be rendered hundreds of times on a page with different sets of data. Still feel free to write a logic for a complicated `lib/` component when needed.
+  - Use named exports (`export const DashboardMenu = () => <div />`), and avoid `default` exports.
+- Naming things:
+  - Always look around the codebase for naming conventions, and follow the best practices of the environment (e.g. use `camelCase` variables in JS, `under_score` in Python).
+  - Use clear, yet functional names (`searchResults` vs `data`).
+  - Logics are camelcase and start with a lowercase (`dashboardLogic`)
+  - React components are camelcase and start with an uppercase (`DashboardMenu`).
```

Comment:
Nit: I'm gonna sound super annoying now, but I think this is called PascalCase IIRC

And to the line above, I think camelCase implies a lowercase first letter

Reply:
Did not search before writing this, so that I open myself to the possibility of being wrong :D

Reply:
Pascal has been introduced to the codebase